exchanges/oex/oex.go

```diff
@@ -0,0 +1,522 @@
+	oexUserInfo        = "/open/api/user/account"
+	oexUserBalanceInfo = "/open/api/user_balance_info"
+	oexAllOrders       = "/open/api/v2/all_order"
+	oexOrderHistory    = "/open/api/all_trade"
+	oexRemoveOrder     = "/open/api/cancel_order"
+	oexRemoveAllOrder  = "/open/api/cancel_order_all"
+	oexCreateOrder     = "/open/api/create_order"
+	oexAllOpenOrders   = "/open/api/v2/new_order"
+	oexSelfTrade       = "/open/api/self_trade"
+	oexFetchOrderInfo  = "/open/api/order_info"
+)
+
+// SetDefaults sets the basic defaults for Oex
+func (o *Oex) SetDefaults() {
+	o.Name = "Oex"
+	o.Enabled = false
+	o.Verbose = false
+	o.RESTPollingDelay = 10
+	o.RequestCurrencyPairFormat.Delimiter = ""
+	o.RequestCurrencyPairFormat.Uppercase = false
+	o.ConfigCurrencyPairFormat.Delimiter = ""
+	o.ConfigCurrencyPairFormat.Uppercase = false
+	o.AssetTypes = []string{ticker.Spot}
+	o.SupportsAutoPairUpdating = false
+	o.SupportsRESTTickerBatching = false
+	o.Requester = request.New(o.Name,
+		request.NewRateLimit(time.Second, 0),
+		request.NewRateLimit(time.Second, 0),
+		common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout))
+	o.APIUrlDefault = oexAPIURL
+	o.APIUrl = o.APIUrlDefault
+	o.Websocket = wshandler.New()
+}
+
+// Setup takes in the supplied exchange configuration details and sets params
+func (o *Oex) Setup(exch *config.ExchangeConfig) {
+	if !exch.Enabled {
+		o.SetEnabled(false)
+	} else {
+		o.Enabled = true
+		o.AuthenticatedAPISupport = exch.AuthenticatedAPISupport
+		o.AuthenticatedWebsocketAPISupport = exch.AuthenticatedWebsocketAPISupport
+		o.SetAPIKeys(exch.APIKey, exch.APISecret, "", false)
+		o.SetHTTPClientTimeout(exch.HTTPTimeout)
+		o.SetHTTPClientUserAgent(exch.HTTPUserAgent)
+		o.RESTPollingDelay = exch.RESTPollingDelay
+		o.Verbose = exch.Verbose
+		o.Websocket.SetWsStatusAndConnection(exch.Websocket)
+		o.BaseCurrencies = exch.BaseCurrencies
+		o.AvailablePairs = exch.AvailablePairs
+		o.EnabledPairs = exch.EnabledPairs
+		err := o.SetCurrencyPairFormat()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = o.SetAssetTypes()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = o.SetAutoPairDefaults()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = o.SetAPIURL(exch)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = o.SetClientProxyAddress(exch.ProxyAddress)
+		if err != nil {
+			log.Fatal(err)
+		}
+		// If the exchange supports websocket, update the below block
+		// err = o.WebsocketSetup(o.WsConnect,
+		//	exch.Name,
+		//	exch.Websocket,
+		//	oexWebsocket,
+		//	exch.WebsocketURL)
+		// if err != nil {
+		// 	log.Fatal(err)
+		// }
+	}
+}
+
+// GetTicker returns ticker info for a selected pair
+func (o *Oex) GetTicker(symbol string) (TickerResponse, error) {
+	var resp TickerResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	path := fmt.Sprintf("%s%s?%s", o.APIUrl, oexGetTicker, params.Encode())
+	err := o.SendHTTPRequest(path, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != "0" {
```

Comment:
Make `"0"` a const variable so if the API ever changes, you can change it there instead of in every single function